From 1e1e9b2283db878e68d8a30cd7a1633a48e83059 Mon Sep 17 00:00:00 2001
From: Ioana Radulescu <ruxandra.radulescu@nxp.com>
Date: Wed, 4 Jan 2017 18:38:23 +0200
Subject: [PATCH 108/767] staging: fsl-dpaa2/eth: Don't use GRO flush timeout

It turns out this causes unnacceptable latency in some
scenarios. So just go back to using napi_complete() instead of
napi_complete_done().

Signed-off-by: Ioana Radulescu <ruxandra.radulescu@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.09.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c
index 474935feaefc..544b08fe7601 100644
--- a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c
+++ b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.c
@@ -1034,7 +1034,7 @@ static int pull_channel(struct dpaa2_eth_channel *ch)
 static int dpaa2_eth_poll(struct napi_struct *napi, int budget)
 {
 	struct dpaa2_eth_channel *ch;
-	int cleaned, rx_cleaned = 0, tx_conf_cleaned = 0;
+	int rx_cleaned = 0, tx_conf_cleaned = 0;
 	bool store_cleaned;
 	struct dpaa2_eth_priv *priv;
 	int err;
@@ -1064,14 +1064,13 @@ static int dpaa2_eth_poll(struct napi_struct *napi, int budget)
 	/* We didn't consume the entire budget, finish napi and
 	 * re-enable data availability notifications
 	 */
-	cleaned = max(rx_cleaned, 1);
-	napi_complete_done(napi, cleaned);
+	napi_complete(napi);
 	do {
 		err = dpaa2_io_service_rearm(NULL, &ch->nctx);
 		cpu_relax();
 	} while (err == -EBUSY);
 
-	return cleaned;
+	return max(rx_cleaned, 1);
 }
 
 static void enable_ch_napi(struct dpaa2_eth_priv *priv)
-- 
2.17.0

